packages/babel-traverse/src/path/lib/hoister.js

```diff
@@ -204,29 +206,38 @@ export default class PathHoister {
     // evaluated more than once leading to a bad optimisation
     if (attachTo.getFunctionParent() === this.path.getFunctionParent()) return;
 
-    // generate declaration and insert it to our point
-    let uid = attachTo.scope.generateUidIdentifier("ref");
-
-    const declarator = t.variableDeclarator(uid, this.path.node);
-
     const insertFn = this.attachAfter ? "insertAfter" : "insertBefore";
-    const [attached] = attachTo[insertFn]([
-      attachTo.isVariableDeclarator()
-        ? declarator
-        : t.variableDeclaration("var", [declarator]),
-    ]);
-
-    const parent = this.path.parentPath;
-    if (parent.isJSXElement() && this.path.container === parent.node.children) {
-      // turning the `span` in `<div><span /></div>` to an expression so we need to wrap it with
-      // an expression container
-      uid = t.JSXExpressionContainer(uid);
-    }
+    if (this.path.isDeclaration()) {
+      const [attached] = attachTo[insertFn](t.cloneNode(this.path.node));
```

Comment:
if we are going to remove this node from the old place right away (line below - `this.path.remove()`) cloning here seems excessive

Reply:
I was worried about reusing the same nodes causing weird scoping issues. Like, the object identity in some scope map would cause it to reference the old path, or something.

Reply:
gotcha 👍